osu.Game.Rulesets.Taiko/Mods/TaikoModSimplified.cs

```diff
@@ -0,0 +1,136 @@
+
+            foreach (var snapConversion in snapConversions)
+            {
+                int patternStartIndex = 0;
+
+                // Skip processing if the corresponding conversion is disabled
+                if (!shouldProcessRhythm(snapConversion.Key))
+                    continue;
+
+                for (int i = 0; i < hits.Count; i++)
+                {
+                    double snapValue = i < hits.Count - 1
+                        ? getSnapBetweenNotes(controlPointInfo, hits[i], hits[i + 1])
+                        : 1; // No next note, default to a safe 1/1 snap
+
+                    if (snapValue == snapConversion.Key)
+                    {
+                        if (!inPattern)
+                        {
+                            patternStartIndex = i;
+                        }
+
+                        inPattern = true;
+                    }
+
+                    // check if end of pattern
+                    if (inPattern && snapValue != snapConversion.Key)
+                    {
+                        // End of the pattern
+                        inPattern = false;
+
+                        // Iterate through the pattern
+                        for (int j = patternStartIndex; j <= i; j++)
+                        {
+                            int currentHitPosition = j - patternStartIndex;
+
+                            if (snapConversion.Key == 8)
+                            {
+                                // 1/8: Remove the second note
+                                if (currentHitPosition % 2 == 1)
+                                {
+                                    toRemove.Add(hits[j]);
+                                }
+                            }
+                            else
+                            {
+                                // 1/6 and 1/3: Adjust the second note and remove the third
+                                if (currentHitPosition % 3 == 1)
+                                {
+                                    hits[j].StartTime = hits[j - 1].StartTime + (controlPointInfo.TimingPointAt(hits[j].StartTime).BeatLength / Convert.ToDouble(snapConversion.Value));
+                                }
+                                else if (currentHitPosition % 3 == 2)
+                                {
+                                    toRemove.Add(hits[j]);
+                                }
+                            }
+                        }
+                    }
+                }
+
+                // Remove queued notes
+                taikoBeatmap.HitObjects.RemoveAll(obj => toRemove.Contains(obj));
+            }
+        }
+
+        private int getSnapBetweenNotes(ControlPointInfo controlPointInfo, Hit currentNote, Hit nextNote)
+        {
+            double gapMs = Math.Max(currentNote.StartTime, nextNote.StartTime) - Math.Min(currentNote.StartTime, nextNote.StartTime);
```

Comment:
Why does this essentially take the absolute difference here, even knowing which note is "current" and which is "next"? Which is to say, why is this not just `nextNote.StartTime - currentNote.StartTime`?

Reply:
original intention was to make it agnostic of order for safety but I guess I'm accounting for an impossible case here lol

will change to what you have